src/ray/raylet/node_manager.cc

```diff
@@ -1435,8 +1435,6 @@ void NodeManager::ProcessDisconnectClientMessage(
 
     // Return the resources that were being used by this worker.
     if (new_scheduler_enabled_) {
-      new_resource_scheduler_->SubtractCPUResourceInstances(
-          worker->GetBorrowedCPUInstances());
```

Comment:
I'm removing these for now, since I saw in some circumstances they were causing a hang in test_args_star_after. They don't seem to be needed for passing the tests.

Perhaps just adding ClearBorrowedCPUInstances() would have been enough; I can also try putting these back if you feel strongly about how we should deal with this in the short term @wuisawesome